### PR TITLE
Documentation for contributing guidelines

### LINKED ISSUE #7: Docs for contributing guidelines
A well-structured guide has to be prepared so that first-time contributors can know the do's and how's of contributing to this very organization.

### PR DESCRIPTION
**Pull request by HargovindArora**
Contribution Guidelines for our Organization
fixes #7

### CODE REVIEW

**Comment by HargovindArora** [comment]
@nemesisdesign Please review this! :) .

File: developer/contributing.rst [hunk 1]
```diff
@@ -1,2 +1,48 @@
 Contributing guidelines
-=======================
\ No newline at end of file
+=======================
+
+We are glad that you choose this organization to give your time to and contribute. Thank you for your interest incontributing to OpenWISP.
```
Comment (xamfy) [inline developer/contributing.rst]: Change choose to chose and give space between in and contributing

File: developer/contributing.rst [hunk 2]
```diff
@@ -1,2 +1,48 @@
...
+If neither of the above works for you please post your views on our `gitter channel. <https://gitter.im/openwisp/general>`_
+
+Steps to Submit your Code or Changes
+------------------------------------
+
+Our main development branch is master it's our central development branch. Your code should be pushed from a different branch and then it is merged into master after proper review from the project maintainers.
+
+1. Create a new branch for your development and a self-descriptive branch name:
+
+     git pull origin master
```
Comment (xamfy) [inline developer/contributing.rst]: make sure these commands appear on different lines

File: developer/contributing.rst (hunk 1 above)
Comment (HargovindArora) [reply developer/contributing.rst]: Ohkay. Thanks

File: developer/contributing.rst [hunk 3]
```diff
@@ -1,2 +1,48 @@
...
+
+2. Commit and push code of your branch:
+
+- Commits should be descriptive in nature, the message should be self-explanatory.
+- Please make sure your code is following our coding style or our project's style. Code should be well formatted.
+- Before committing and pushing the changes , test the code on your local machine first.
+- After pushing your branch code, make a Pull Request of that corresponding change of yours which should contain a descriptive meassage and mention the issue number too(if there is one). Example- "Fixes #123".
+
+      git commit -m "your message here"
+      git push <remote name> <branch name>
```
Comment (xamfy) [inline developer/contributing.rst]: same here

File: developer/contributing.rst (hunk 2 above)
Comment (HargovindArora) [reply developer/contributing.rst]: Yeah. Doing it

**Comment by xamfy** [comment]
@HargovindArora please view this file on your local machine in your browser to see if the commands are displayed properly.

**Comment by HargovindArora** [comment]
Yeah Sure. By sharing their experiences it will promote the organization. And we can introduce them to Committing style through autopep8 for python and mentioning about non-technical stuff is also a good idea!
Until this is done should we push the current doc so that people can get to know at least some of the things about how they should get started @nemesisdesign?

**Comment by nemesisdesign** [comment]
@HargovindArora perfect, just keep in mind we don't use `autopep8`, we use `flake8` and `isort` as django does, we follow their same conventions on the python side.

**Comment by HargovindArora** [comment]
Oh. Ohkay. Got it!

**Comment by nemesisdesign** [comment]
@HargovindArora sorry I missed your point about pushing these changes now, well for me is fine but are you really sure this kind of information is going to tell anything useful to programmers who want to contribute to openwisp? Most of them already know how to commit and create branches

**Comment by HargovindArora** [comment]
@nemesisdesign Yeah you're right that most of the people might be knowing that basic stuff but we have created a generalized doc so that even if there is a newbie who wants to get started, he/she can get some starting points or some push from our doc, I mean that the doc should be for everyone! :)

**Pull request merged**